src/algebra/monoid_algebra.lean

```diff
@@ -1170,6 +1170,60 @@ See note [partially-applied ext lemmas]. -/
   f = g :=
 ring_hom_ext (ring_hom.congr_fun h₁) (monoid_hom.congr_fun h_of)
 
+section opposite
+
+variables {I : Type*} [semiring R]
+
+open finsupp opposite
+
+-- simps doesn't do well here, but we could add some lemmas
+-- about `support` and `coe_fn` that are true by rfl
+/-- The opposite of an `add_monoid_algebra R I` is additively equivalent to
+the `add_monoid_algebra Rᵒᵖ I` over the opposite ring, taking elements to their opposite. -/
+protected noncomputable def op_add_equiv :
+  (add_monoid_algebra R I)ᵒᵖ ≃+ add_monoid_algebra Rᵒᵖ I :=
+opposite.op_add_equiv.symm.trans
+{ to_fun := λ p, ⟨p.support, op ∘ (p : I → R), λ i, mem_support_iff.trans (op_ne_zero_iff _).symm⟩,
+  inv_fun := λ p, ⟨p.support, unop ∘ p, λ i, mem_support_iff.trans (unop_ne_zero_iff _).symm⟩,
```

Comment:
It's definitionally marginally worse (doesn't preserve the support by rfl), but `finsupp.map_range.add_equiv opposite.op_add_equiv` or similar can replace those manual construction